src/Engine/ProtoAssociative/CodeGen.cs

```diff
@@ -8560,6 +8561,20 @@ Assert unhandled
                             symbolnode.SetStaticType(castType);
                         }
                         castType = symbolnode.staticType;
+
+
+                        if (bnode.IsInputExpression)
+                        {
+                            StackValue regLX = StackValue.BuildRegister(Registers.LX);
+                            EmitInstrConsole(ProtoCore.DSASM.kw.pop, ProtoCore.DSASM.kw.regLX);
+                            EmitPop(regLX, globalClassIndex);
+
+                            graphNode.updateBlock.updateStartPC = pc;
+
+                            EmitInstrConsole(ProtoCore.DSASM.kw.push, ProtoCore.DSASM.kw.regLX);
+                            EmitPush(regLX);
+                        }
```

Comment:
Yes, this ensures that on an input expression, they get temporarily stored in LX